flask/app/users.py

```diff
@@ -24,3 +27,84 @@ def list_users():
             for user in users
         ]
     )
+
+
+@app.route("/api/users/<string:username>", methods=["GET"])
+@login_required
+def get_user(username: str):
+    if (
+        not app.config.get("LOGIN_DISABLED")
+        and not current_user.is_admin
+        and username != current_user.username
+    ):
+        return "Unauthorized", 401
+
+    user = (
+        models.User.query.options(joinedload(models.User.groups))
+        .filter(models.User.username == username)
+        .first_or_404()
+    )
+
+    return jsonify(
+        {
+            "username": user.username,
+            "email": user.email,
+            "is_admin": user.is_admin,
+            "last_login": user.last_login,
+            "deactivated": user.deactivated,
+            "groups": [group.group_code for group in user.groups],
+            "minio_access_key": user.minio_access_key,
+            "minio_secret_key": user.minio_secret_key,
+        }
+    )
+
+
+@app.route("/api/users/<string:username>", methods=["POST"])
+@login_required
+def reset_minio_user(username: str):
+    if (
+        not app.config.get("LOGIN_DISABLED")
+        and not current_user.is_admin
+        and username != current_user.username
+    ):
+        return "Unauthorized", 401
+
+    if not request.json:
+        return "Request body must be JSON", 415
```

Comment:
Is there a reason we care about the body at all?

Reply:
We don't really and we can go through later and change all these instances where the request body doesn't matter. Instead of this check, which also fails if the body is `{}` or `[]` we can also check the Content-Type instead. My reasoning for this is that regular HTML forms can perform GET and POST requests cross-domain, but they can only submit `application/x-www-form-urlencoded` and `multipart/form-data`, not JSON, so it's just a way to deny those types of API accesses.

Reply:
Ohhh I see... yeah we should leave replacing most pointless checks with checking Content-Type header for a future sprint